src/test/java/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(classes = OnlineBookstoreApplication.class)
+@Sql(scripts = "classpath:database/books/add-three-default-books.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+class BookControllerTest {
+
+    private static MockMvc mockMvc;
+
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookService bookService;
+
+    @BeforeEach
+    void setUp(@Autowired WebApplicationContext applicationContext) {
+        objectMapper = new ObjectMapper();
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get all books")
+    void getAllBooks_ReturnsAllBooks_ExpectedSuccess() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/books")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .param("page", "0")
+                                .param("size", "5")
+                )
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.content").isArray())
+                .andReturn();
+
+        assertNotNull(result.getResponse().getContentAsString());
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("Get book by ID")
+    void getBookById_ReturnsBookById_ExpectedSuccess() throws Exception {
+        MvcResult result = mockMvc.perform(
+                        get("/books/{id}", 1L)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actualBook = objectMapper.readValue(
+                result.getResponse().getContentAsByteArray(), BookDto.class);
+        assertNotNull(actualBook);
+        assertEquals(1L, actualBook.getId());
+    }
+
+    @Sql(scripts = "classpath:database/categories/add-fantazy-category-to-categories-table.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new book")
+    void createBook_CreatesNewBook_ExpectedSuccess() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setAuthor("John Doe");
+        requestDto.setTitle("New Book Title");
+        requestDto.setDescription("Book Description");
+        requestDto.setIsbn("4" + System.currentTimeMillis());
+        requestDto.setCoverImage("http://example.com/cover3.jpg");
+        requestDto.setPrice(BigDecimal.valueOf(15));
+        Category category = new Category();
+        category.setId(1L);
```

Comment:
Move it to the private method and don't use `System.currentTimeMillis()` for isbn generation